### PR TITLE
Added node driver resource

### PR DESCRIPTION
**Pull request by rodcloutier**
Added new resource for Node Driver

### CODE REVIEW

**Comment by rawmind0** [comment]
Hi @rodcloutier . Thanks for the PR including node-driver.

Could you please include doc file about this resource on https://github.com/rancher/terraform-provider-rancher2/tree/master/website/docs/r folder and reference it from https://github.com/rancher/terraform-provider-rancher2/blob/master/website/rancher2.erb file??

**Comment by rodcloutier** [comment]
Sure. I should have thought about this.

**Comment by rodcloutier** [comment]
@rawmind0 I added the documentation for the node driver resource.

File: rancher2/resource_rancher2_node_driver.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"whitelist_domains": &schema.Schema{
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_driver.go:54]: Could you please add `annotations` and `labels` fields??

```
+		"annotations": &schema.Schema{
+			Type:        schema.TypeMap,
+			Optional:    true,
+			Computed:    true,
+			Description: descriptions["annotations"],
+		},
+		"labels": &schema.Schema{
+			Type:        schema.TypeMap,
+			Optional:    true,
+			Computed:    true,
+			Description: descriptions["labels"],
+		},
```

File: rancher2/resource_rancher2_node_driver.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+	err = d.Set("url", in.URL)
+	if err != nil {
+		return err
+	}
+
+	err = d.Set("whitelist_domains", toArrayInterface(in.WhitelistDomains))
+	if err != nil {
+		return err
+	}
+
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_driver.go:123]: `annotations` and `labels` fields on flatten

```
+	err = d.Set("annotations", toMapInterface(in.Annotations))
+	if err != nil {
+		return err
+	}
+
+	err = d.Set("labels", toMapInterface(in.Labels))
+	if err != nil {
+		return err
+	}
+
```

File: rancher2/resource_rancher2_node_driver.go [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+	obj.Checksum = in.Get("checksum").(string)
+	obj.Description = in.Get("description").(string)
+	obj.ExternalID = in.Get("external_id").(string)
+	obj.Name = in.Get("name").(string)
+	obj.UIURL = in.Get("ui_url").(string)
+	obj.URL = in.Get("url").(string)
+
+	if v, ok := in.Get("whitelist_domains").([]interface{}); ok && len(v) > 0 {
+		obj.WhitelistDomains = toArrayString(v)
+	}
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_driver.go:161]: `annotations` and `labels` fields on expander

```
+
+	if v, ok := in.Get("annotations").(map[string]interface{}); ok && len(v) > 0 {
+		obj.Annotations = toMapString(v)
+	}
+
+	if v, ok := in.Get("labels").(map[string]interface{}); ok && len(v) > 0 {
+		obj.Labels = toMapString(v)
+	}
+
```

File: rancher2/resource_rancher2_node_driver.go [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+	update := map[string]interface{}{
+		"active":           d.Get("active").(bool),
+		"builtin":          d.Get("builtin").(bool),
+		"checksum":         d.Get("checksum").(string),
+		"description":      d.Get("description").(string),
+		"externalId":       d.Get("external_id").(string),
+		"name":             d.Get("name").(string),
+		"uiUrl":            d.Get("ui_url").(string),
+		"url":              d.Get("url").(string),
+		"whitelistDomains": toArrayString(d.Get("whitelist_domains").([]interface{})),
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_driver.go:247]: `annotations` and `labels` fields on update

```
+		"annotations":   toMapString(d.Get("annotations").(map[string]interface{})),
+		"labels":        toMapString(d.Get("labels").(map[string]interface{})),
```

File: website/docs/r/nodeDriver.html.markdown [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+* `active` - (Required) Specify if the node driver state.
+* `builtin` - (Required) Specify wheter the node driver is an internal node driver or not.
+* `checksum` - (Optional) Verify that the downloaded driver matches the expected checksum.
+* `description` - (Optional) Description of the node driver.
+* `external_id` - (Optional) External ID
+* `name` - (Required) Name of the node driver.
+* `ui_url` - (Optional) The URL to load for customized Add Nodes screen for this driver.
+* `url` - (Required) The URL to download the machine driver binary for 64-bit Linux.
+* `whitelist_domains` - (Optional) Domains to whitelist for the ui.
```
Comment (rawmind0) [inline website/docs/r/nodeDriver.html.markdown:42]: `annotations` and `labels` fields on doc

```
+* `annotations` - (Optional/Computed) Annotations of the resource (map).
+* `labels` - (Optional/Computed) Labels of the resource (map).
```

**Review by rawmind0 — CHANGES_REQUESTED** [review]

**Comment by rawmind0** [comment]
@rodcloutier thanks for adding documentation on the resource. 

Could you please take a look to the changes requested?

File: rancher2/resource_rancher2_node_driver.go (hunk 4 above)
Comment (rodcloutier) [reply rancher2/resource_rancher2_node_driver.go:247]: Done

**Comment by rodcloutier** [comment]
I have added the `annotations` and `labels` fields.

**Review by rawmind0 — APPROVED** [review]

**Comment by rawmind0** [comment]
Thanks @rodcloutier 

LGTM

**Pull request merged**